### ISSUE TITLE
Ability to define custom commands

### ISSUE THREAD

**Issue by dpc** [ipetkov/crane#301]
This is an extension/alternative of #300 

Ofentime `crane.buildPackage` or other command does something slightly not the way I want it to, and usually when I ask about it, the answer is "just use mkCargoDerivation" directly. Which is fair but has certain negative consequences.

Lots of my craneLib calls are now `mkCargoDerivation` with more arguments, and they don't carry semantic meaning of `craneLib.cargoTest` or `craneLib.buildPackage`. Please note that I usually maintain `flake.nix` for projects with teams of developers of various (usually  far from expert) Nix skills and knowledge, so stuff like that is important so that people can navigate and debug CI and other scripts and tooling.

So, to the chase:

If I could define custom commands for `craneLib` as wrappers around existing commands, I could make my own crane commands that I need. They could also carry extra logic somewhat line in #300 .

I could then have:

```
craneLib.buildPkgsCrossCompile { ... };
craneLib.cargoCustomTest { ... };
```

etc.

**Comment by ipetkov** [comment]
Similar to [my comment here](https://github.com/ipetkov/crane/issues/300#issuecomment-1509985673) you can use `overrideScope'` to inject your own definitions if you really want them available inside of `craneLib`:

```nix
let
  craneLib' = (crane.mkLib pkgs).overrideScope' (final: prev: {
    myWhatever = args: prev.buildPackage (args // { pname = "whatever"; });
  });
in
craneLib'.myWhatever {
  src = ...;
}
```

Would this achieve what you are looking for?

**Comment by ipetkov** [comment]
I've documented using `overrideScope'` in #303, though it's also entirely possible to add your own wrapper around `craneLib` which includes additional commands specific to your project without using it directly.

I'm going to consider this resolved with #303 since I don't think there is anything that `crane` does which prevents downstream consumers from adding their own wrappers, but feel free to reopen if you have additional followups!
